### PR TITLE
Fix garbage_check not preserving image IDs

### PR DESCRIPTION
**Pull request by cevich**
* Additional regex magic & processing is needed when a FQIN is specified
without the repo-address component.
* Images module now uses key-words to reference components of FQIN
group matches
* Added new unittests to verify above changes
* Updated garbage_check to be more flexible when comparing against
configured images to preserve.

### CODE REVIEW

File: dockertest/images.py [hunk 1]
```diff
@@ -54,8 +54,11 @@ class DockerImage(object):  # pylint: disable=R0902
     #: parsing, spec defined in docker-io documentation.  e.g.
     #: ``[registry_hostname[:port]/][user_name/]
     #: (repository_name[:version_tag])``
-    repo_split_p = re.compile(r"(.+?(:[\d]+?)?/)?([\w\-\.\+]+/)?"
-                              r"([\w\-\.]+)(:[\w\-\.]+)?")
+    repo_split_p = re.compile(r"(?P<repo_addr>.+?"  # optional
```
Comment (edsantiago) [inline dockertest/images.py]: If you add an extra layer of parens:

     r"((?P<repo_addr> [...] )?)/)?"  # new paren after r" and immediately before the slash


...you can remove the awkward trim-slash at line 145-146.

File: dockertest/images.py [hunk 2]
```diff
@@ -136,15 +139,26 @@ def split_to_component(cls, full_name):
...
-            if user:
-                user = user[:-1]  # remove trailing slash
-            if tag:
-                tag = tag[1:]  # remove beginning colon
-        except TypeError:  # no match
+            full_name = full_name.strip()
+            mobj = cls.repo_split_p.match(full_name)
+            repo_addr = mobj.group('repo_addr')
+            if repo_addr is not None:
+                repo_addr = repo_addr.replace('/', '')
```
Comment (edsantiago) [inline dockertest/images.py]: See above.

File: dockertest/images.py [hunk 3]
```diff
@@ -54,8 +54,11 @@ class DockerImage(object):  # pylint: disable=R0902
     #: parsing, spec defined in docker-io documentation.  e.g.
     #: ``[registry_hostname[:port]/][user_name/]
     #: (repository_name[:version_tag])``
-    repo_split_p = re.compile(r"(.+?(:[\d]+?)?/)?([\w\-\.\+]+/)?"
-                              r"([\w\-\.]+)(:[\w\-\.]+)?")
+    repo_split_p = re.compile(r"(?P<repo_addr>.+?"  # optional
+                              r"(?P<repo_addr_port>:[\d]+?)?/)?"  # optional
+                              r"(?P<user>[\w\-\.\+]+/)?"  # optional
```
Comment (edsantiago) [inline dockertest/images.py]: Likewise: ((?P<user> ... )/)?

File: dockertest/images.py [hunk 4]
```diff
@@ -136,15 +139,26 @@ def split_to_component(cls, full_name):
...
-                tag = tag[1:]  # remove beginning colon
-        except TypeError:  # no match
+            full_name = full_name.strip()
+            mobj = cls.repo_split_p.match(full_name)
+            repo_addr = mobj.group('repo_addr')
+            if repo_addr is not None:
+                repo_addr = repo_addr.replace('/', '')
+            user = mobj.group('user')
+            if user is not None:
+                user = user.replace('/', '')
```
Comment (edsantiago) [inline dockertest/images.py]: Likewise.

**Comment by cevich** [comment]
@edsantiago Good idea!  Bot I miss someone(s) looking over my shoulder.    Unf. I'm in a bit of a rush with this change (for CI's sake) and I *think* what's here now is 'working'.   It's a case of ugly-but-working > pretty-but-need-to-re-test.  In fact, I've squeezed in testing along with https://github.com/cevich/autotest-docker-enabled-product-testing/pull/26 to try and save some time...uggg...so rushed!  Though I'll open an issue on your suggestion, because it's a good one.

**Pull request merged**